### ISSUE TITLE
Configuration of Y Lights

### ISSUE THREAD

**Issue by FidgetyRat** [fu-raz/signalrgb-govee-direct-connect#3]
I have the Y lights from Govee that do work with this component, but not quite correct.  Signal seems to treat the entire layout as only 15 LEDs laid out horizontally where led 1 may light the left arm of 2 panels. LED 2 the top of one. Etc. these panels have at least 9 LEDs per panel times 7 panels. 

Any chance of better support for these panels? While it works with only 15 LEDs it’s lacking a lot of the granularity the Govee app has in addressing individual LEDs vs treating large chunks of each panel as a single led. 

I tried duplicate, custom component, etc.

**Comment by fu-raz** [comment]
If you say there are 9 leds per panel for 7 panels you should enter 9*7=63 leds in the plugin. Then I would suggest choosing components. All you need then is a component JSON for a panel. We could try to figure out the led layout together. Can you send a picture of one of the panels? Preferably turned on so I can see the placements of the LEDs?

**Comment by fu-raz** [comment]
![image](https://github.com/user-attachments/assets/21b74e63-21aa-451e-b7dd-8396ba926961)
Something like this? But the LED index is probably wrong

Put this in your Documents/WhirlwindFX/Components directory and select `Govee Y-Light` as the component
[Y-Light.json](https://github.com/user-attachments/files/18247219/Y-Light.json)

**Comment by FidgetyRat** [comment]
Thanks for the reply. I took some photos/videos. It's got some strange behavior, that's for sure. 

To start, I closed Signal and opened the device in the Govee phone app and turned on 2 of the panels with two separate colors. The app doesn't give you pixel per pixel granularity, just entire panels for this mode. I still to this day can't find any info online as to how many actual LEDs are in each Y panel, you would think they would put that info SOMEWHERE and damn did they do a good job diffusing them to the point I have trouble seeing individuals.
![IMG_6946](https://github.com/user-attachments/assets/bd567423-25f8-449a-8d79-9ea46890683c)

The video is one of the Govee "scenes". Here you see they do have individual control of pixels, so it's not a full panel situation. 
https://github.com/user-attachments/assets/c7ea0fd1-d5d2-47b8-a09f-fa79523f3c7e

Finally, I added your component, set the device to 63 LEDs (my best guess) and added a single 9 LED component. When I light up that component, it's lighting several full panels. 
![IMG_6948](https://github.com/user-attachments/assets/e3f5ceaa-b3b3-4df4-b40d-75af7caf55ae)

The best I have been able to do was to define a "single device" with 15 LEDs (15 seems to be the max signal will ever allow to light). In this mode, it creates a strip of 15 bars on layout and the entire wall design is treated as 15 vertical zones.

On a side note, as of recent Signal updates, these lights never reconnect anymore when the PC is woken up. I have to restart signal every time. They used to reconnect within a few seconds of waking.

**Comment by fu-raz** [comment]
When I watch the video and try to count the leds I count  about 12 per bar, so that would be 36 per unit. If we want to know for sure you should try the following:

Only connect 1 Y light and make sure the app knows it's only one light. Then using my plugin, try a 'single device' starting with 30 leds. Then go into the device settings (device -> go to settings):
![image](https://github.com/user-attachments/assets/1dbf3539-7b01-4e99-99fb-e9d7e02c26a8)

Then click on the paint button to start led painting:
![image](https://github.com/user-attachments/assets/24bdcf21-2398-4e52-9fd4-ce2f733abde0)

Then click on the brush and make a pattern like this:
![image](https://github.com/user-attachments/assets/f5bfb2dc-f0e1-4625-8e93-e5dcd4b37975)

Then look at your device on the wall and you should probably see the pattern. Then count the amount of red leds you see. If it's less than the amount of red leds you 'painted' in SRGB, then you should lower the total amount of leds and try again.

If it is the same amount, check if you see some 'stretched' leds. These devices tend to stretch the pattern if there are more lights than you set the plugin to. For example: 

If I have a device with 6 leds and I set the amount of leds to 4. Then I paint the leds in SRGB like this XOXO where X would be red and O would be black. On the device it would probably look like this: XXOXOO. If that makes sense.

Also:, I'll look into the problem with waking from sleep.

**Comment by fu-raz** [comment]
To make it easier for you to test, I added a new feature. You should make sure you're on the main branch of the plugin:
![image](https://github.com/user-attachments/assets/9cd6ba0c-cfb5-4620-aa3a-86bbc74729f8)

You should make sure you set the amound of leds high. I would say 12 x the amount of Y lights you have to start.

Then in the device settings' Lighting tab you'll be able to set lighting mode to: "Test pattern", after that you'll be able to play around with the amount of leds using the slider.

![image](https://github.com/user-attachments/assets/e920b1df-72a9-4824-b263-d0667943807a)

Start the slider at 1 and then slide to the right. Stop when the lights on the device no longer move. That should be the right total amount of leds for the devices. Then we divide that by the amount of devices you have and we should get the right count for 1 Y light.

**Comment by fu-raz** [comment]
I've just discovered there's a new protocol and I'm adding it to the plugin. This might solve the problem

**Comment by fu-raz** [comment]
I've just pushed changes to the main branch. This should probably fix the problem!

**Comment by FidgetyRat** [comment]
I updated and noticed the Dreamview V2 protocol. If selected, the lights do nothing (locks up) so I guess my device doesn't support. If I select the classic Dreamview it treats each panel as a single LED and if I select Razer it's the case where the device is broken into 15 vertical stripes.

**Comment by fu-raz** [comment]
> I updated and noticed the Dreamview V2 protocol. If selected, the lights do nothing (locks up) so I guess my device doesn't support. If I select the classic Dreamview it treats each panel as a single LED and if I select Razer it's the case where the device is broken into 15 vertical stripes.

Have you tried what I said before? Just set the led count really high and then use the test pattern and the test led count slider to see what happens when you change the led count?

**Comment by kev368** [comment]
not sure if this every got resolved   but each y light has 4 leds in each leg . looking forward to trying this add on .

**Comment by FidgetyRat** [comment]
> Have you tried what I said before? Just set the led count really high and then use the test pattern and the test led count slider to see what happens when you change the led count?

> Have you tried what I said before? Just set the led count really high and then use the test pattern and the test led count slider to see what happens when you change the led count?

Yeah I have tried that. I normally have the LEDs set at 63 (assumed 3 per leg, 9 per tile, times 7 tiles) and with razer driver signal only ever sees 15 LEDs, basically 15 vertical stripes along the pattern from left to right even if divided using the custom devices. 

Dreamview treats each full panel as a single led. 

It’s strange. I assume the API built into the device just doesn’t support full control of individual LEDs.

**Comment by fu-raz** [comment]
I guess it doesn't, really hoped the last update would've fixed it. Cause there was this govee string light that had the same sort of problem. If you set the led count to the amount of leds (segments) the string has, it would just show 10. Which ended up being a bug in my plugin.
